File: src/main/java/com/github/devcsrj/isdue/api/Invoice.java
package com.github.devcsrj.isdue.api;

import org.joda.money.Money;

import java.time.ZonedDateTime;
import java.util.List;

public record Invoice(
        String id,
        Biller biller,
        String accountNumber,
        ZonedDateTime issuedAt,
        ZonedDateTime dueAt,
        Money total,
        List<Item> items) {

    public record Item(
            String description,
            ZonedDateTime date,
            Money amount) {
    }
}

File: api/src/main/kotlin/com/github/devcsrj/isdue/Invoice.kt
/**
 * Is Due | API - Your one-stop app for managing Philippine service provider invoices
 * Copyright © 2017 Reijhanniel Jearl Campos (devcsrj@apache.org)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
package com.github.devcsrj.isdue

import java.math.BigDecimal
import java.time.LocalDate
import java.time.ZonedDateTime

data class Invoice(
        val id: String,
        val amount: BigDecimal,
        val date: LocalDate
)